pytket_qir/converter.py

```diff
@@ -0,0 +1,357 @@
+            curr_qir_block = cast(
+                QirBlock, self.module_function.get_block_by_name(block_name)
+            )
+            circuit = Circuit(self.parser.qubits, self.parser.bits + 2)
+            # Setting the condition True for each block.
+            circuit.add_c_setbits([1], [self.parser.bits])
+            condition_bit = circuit.get_c_register("c")[self.parser.bits]
+            self.conditions[block_name] = condition_bit
+            for compo_block in block.composition:
+                qir_block = cast(
+                    QirBlock, self.module_function.get_block_by_name(compo_block)
+                )
+                new_circuit = self.parser.block_to_circuit(
+                    qir_block, Circuit(self.parser.qubits, self.parser.bits + 2)
+                )
+                circuit.append(new_circuit)
+            term = curr_qir_block.terminator
+            if isinstance(term, QirCondBrTerminator):
+                term_condition = cast(QirLocalOperand, term.condition)
+                condition_name = "%" + str(term_condition.name)
+                # Retrieving the condition bit from the circuit log.
+                self.local_conditions[block_name] = self.parser.ssa_vars[condition_name]
+            elif isinstance(term, (QirBrTerminator, QirRetTerminator)):
+                # Return is considered an unconditional branch.
+                self.local_conditions[block_name] = condition_bit
+            preds = block.preds
+            if not preds:  # Entry block.
+                expr = self.local_conditions["entry_pred"]
+                self.conditions[block_name] = self.conditions[block_name] | (
+                    self.conditions["entry_pred"] & expr
+                )
+            else:
+                # import pdb; pdb.set_trace()
+                for pred in preds:
+                    if (
+                        self.edges[pred][block_name] == None
+                        or self.edges[pred][block_name] == True
+                    ):
+                        expr = self.local_conditions[pred]
+                    elif self.edges[pred][block_name] == False:
+                        expr = BitNot(self.local_conditions[pred])
+                    self.conditions[block_name] = self.conditions[block_name] | (
+                        self.conditions[pred] & expr
+                    )
+
+            # Add extra bits created in the sub-circuit to the main one.
+            for bit in set(circuit.bits) - set(main_circuit.bits):
+                main_circuit.add_bit(bit)
+            circ_box = CircBox(circuit)
+            arguments = [qubit.index[0] for qubit in circuit.qubits] + [
+                bit.index[0] for bit in circuit.bits
+            ]  # Order matters.
+            main_circuit.add_circbox(
+                circ_box, arguments, condition=if_bit(self.conditions[block_name])
+            )
+        return main_circuit
+
+    def apply_contraction(self, block) -> Block:
```

Comment:
This is doing the fusion of two blocks? Or?

Reply:
Yes exactly. I've added a comment.